src/bgw/job.c

```diff
@@ -1054,6 +1056,56 @@ zero_guc(const char *guc_name)
 				(errcode(ERRCODE_INTERNAL_ERROR), errmsg("could not set \"%s\" guc", guc_name)));
 }
 
+extern Oid
+ts_bgw_job_get_funcid(BgwJob *job)
+{
+	ObjectWithArgs *object = makeNode(ObjectWithArgs);
+	object->objname = list_make2(makeString(NameStr(job->fd.proc_schema)),
+								 makeString(NameStr(job->fd.proc_name)));
+	object->objargs = list_make2(SystemTypeName("int4"), SystemTypeName("jsonb"));
+
+	return LookupFuncWithArgs(OBJECT_ROUTINE, object, false);
+}
+
+static char *
+ts_bgw_job_function_call_string(BgwJob *job)
```

Comment:
Can you add a test that exercises this function? For example, create a PG function that accepts a job id and fetches the call string for the function. It would be good to verify that it produces what it is supposed to produce.

Reply:
You're correct. Yesterday when I finished to polish it and push new commit I got uncomfortable with the lack of coverage in this function. Will add some tests.